src/OrbitGl/TimelineTicks.h

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_GL_TIMELINE_IMPL_H_
+#define ORBIT_GL_TIMELINE_IMPL_H_
+
+#include <cstdint>
+#include <set>
+#include <vector>
+
+namespace orbit_gl {
+
+constexpr uint64_t kSecondToNano = 1000 * 1000 * 1000;
+constexpr uint64_t kMinuteToNano = 60 * kSecondToNano;
+constexpr uint64_t kHourToNano = 60 * kMinuteToNano;
+
+class TimelineTicks {
+  enum class TickType { MajorTick, MinorTick };
+
+ public:
+  TimelineTicks();
+
+  // For now, only major ticks.
+  [[nodiscard]] std::vector<std::pair<TickType, uint64_t> > GetTicks(uint64_t start_ns,
+                                                                     uint64_t end_ns) const;
+  [[nodiscard]] std::vector<uint64_t> GetMajorTicks(uint64_t start_ns, uint64_t end_ns) const;
+
+ private:
+  uint64_t GetScale(uint64_t time_range_ns) const;
+  std::set<uint64_t> scales_;
```

Comment:
I would prefer `std::chrono::nanoseconds` instead of a plain `uint64_t` in all those function arguments and potentially return values.

Reply:
This would also avoid the need for the `kSecondToNano` constant since chrono provides duration_cast for that.

Reply:
I would favor consistency, I believe we use uint64_t as timestamps in nanoseconds in all other places.

Reply:
Yes, we are using uint64_t almost everywhere but we use Duration in DisplayFormats. I will think about that, it might make the code more readable.

Reply:
As discussed offline with @beckerhe, I will keep them uint64_t for now. We could consider in the future migrate everything to use std::chrono.